lib/browser/api/ipc-main.ts

```diff
@@ -1,8 +1,44 @@
 import { EventEmitter } from 'events'
+import { IpcMainEvent } from 'electron'
 
-const emitter = new EventEmitter()
+class IpcMain extends EventEmitter {
+  _invokeHandlers: {[channel: string]: (e: IpcMainEvent, ...args: any[]) => void};
```

Comment:
```suggestion
  private _invokeHandlers: Record<string, (e: IpcMainEvent, ...args: any[]) => Promise<any>> = {};
```

Reply:
made it a `Map` instead. These functions don't return `Promise`s, since they wrap `event.reply`/`event.throw`.